src/protocol/vm/adapter_contract.rs

```diff
@@ -199,17 +199,17 @@ where
                             .into_uint()
                             .unwrap();
                         if denominator.is_zero() {
-                            Err(SimulationError::DecodingError("Denominator is zero".to_string()))
+                            Err(SimulationError::FatalError("Denominator is zero".to_string()))
                         } else {
                             Ok((numerator.as_u128() as f64) / (denominator.as_u128() as f64))
                         }
                     } else {
-                        Err(SimulationError::DecodingError("Invalid fraction tuple".to_string()))
+                        Err(SimulationError::FatalError("Invalid fraction tuple".to_string()))
                     }
                 })
                 .collect()
         } else {
-            Err(SimulationError::DecodingError("Price is not a Token::Array".to_string()))
+            Err(SimulationError::FatalError("Price is not a Token::Array".to_string()))
```

Comment:
Can we print what the raw value of the price if it does not fit into the type we expected?